### PR TITLE
chore: handle case of empty child filter

### PR DESCRIPTION
**Pull request by saxenakshitiz**

### CODE REVIEW

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java [hunk 1]
```diff
@@ -136,10 +136,16 @@ private String convertFilterToString(
...
       builder.append("( ");
       for (Filter childFilter : filter.getChildFilterList()) {
-        builder.append(delim);
-        builder.append(convertFilterToString(childFilter, paramsBuilder, executionContext));
-        builder.append(" ");
-        delim = operator + " ";
+        // child filter should either have further child filters or should have LHS and RHS
+        // only then child filter is converted to string
+        // this check avoids converting empty filters to string
+        if (!childFilter.getChildFilterList().isEmpty()
```
Comment (satish-mittal) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java:142]: These checks should go as part of query transformation that happens before actual query execution. See `AbstractQueryTransformation.transform()`. There is already transformFilter() that skips certain defaults.

Comment (kotharironak) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java:142]: +1

File: query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.stream.Collectors;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+public abstract class AbstractServiceTest<TQueryServiceRequestType extends GeneratedMessageV3> {
+
+  private static final String QUERY_SERVICE_TEST_REQUESTS_DIR = "test-requests";
+
+  protected TQueryServiceRequestType readQueryServiceRequest(String fileName) {
```
Comment (kotharironak) [inline query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java:17]: What is `T`stands for? What about just having `QueryServiceRequestType` Is it how it was named in `proto`?

File: query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    String resourceFileName =
+        createResourceFileName(
+            QUERY_SERVICE_TEST_REQUESTS_DIR,
+            fileName); // QUERY_SERVICE_TEST_REQUESTS_DIR + FILE_SEPARATOR + getTestSuiteName() +
```
Comment (kotharironak) [inline query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java:21]: nit: shall we remove this comment? The function `createResourceFileName` internally explain this.

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java (hunk 1 above)
Comment (saxenakshitiz) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java]: done

File: query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java (hunk 3 above)
Comment (saxenakshitiz) [reply query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java:21]: done

File: query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java (hunk 2 above)
Comment (saxenakshitiz) [reply query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java:17]: I think it is to denote a generic class type. It is same as definer here by you - https://github.com/hypertrace/gateway-service/blob/1997536bcdddfc4d00aed3af7392fc1bb2ecdf91/gateway-service-impl/src/test/java/org/hypertrace/gateway/service/common/AbstractServiceTest.java#L45

**Review by satish-mittal — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]

**Pull request merged**